proxy/ProxyClientTransaction.h

```diff
@@ -211,7 +211,11 @@ class ProxyClientTransaction : public VConnection
 
   virtual bool allow_half_open() const = 0;
 
-  virtual const char *get_protocol_string() const = 0;
+  virtual const char *
```

Comment:
I don't think adding a length argument makes much difference one way or the other.  I guess it would reduce the logging call site to once call.  But we are storing the protocol as a NULL terminated string, so we'd be doing the strlen in any case.